Объединены Country и CountryModel.

Denis7God opened the pull request:
- Вместо создания CountryModel, вне контекста создаются объекты Country. При добавлении в избранное, объект перемещается в главный контекст и сохраняется (CountryInfoViewController, метод favoriteBarButtonPressed).
- Добавлены комментарии к некоторым элементам.

Denis7God committed:
Объединены Country и CountryModel.

- Вместо создания CountryModel, вне контекста создаются объекты Country. При добавлении в избранное, объект перемещается в главный контекст и сохраняется (CountryInfoViewController, метод favoriteBarButtonPressed).
- Добавлены комментарии к некоторым элементам.

OleksiiKarpenko commented on Atlas/Controller/CountryInfoViewController.swift:
```diff
@@ -11,7 +11,7 @@ import CoreData
 
-    var country: CountryModel!
+    var country: Country!
     var countryInfo: CountryInfo!
```
Это что за анврапы, а?

OleksiiKarpenko commented on Atlas/Controller/CountryInfoViewController.swift:
```diff
@@ -36,51 +36,52 @@ class CountryInfoViewController: UIViewController, UITableViewDelegate, UITableV
-            let newCountryObject = Country(countryModel: country)
-            context.insert(newCountryObject)
+            context.insert(self.country)
             try? context.save()
```
А кетч где?)

OleksiiKarpenko commented on Atlas/Model/Country+CoreDataClass.swift:
```diff
@@ -9,19 +9,30 @@
+              let nativeName = json["nativeName"],
+              let alpha2Code = json["alpha2Code"],
+              let alpha3Code = json["alpha3Code"]
+        else {
```
Хм, может тут лучше какие-то модельки кидать, а не стринги?

OleksiiKarpenko commented on Atlas/Model/CountryInfo.swift:
```diff
@@ -15,7 +15,8 @@ struct CountryInfo {
     let borders: [String]
     let latitude: Float
     let longitude: Float
-
```
Лучше дабл использовать. Кстати почему лучше дабл?

Denis7God replied:
Точность у дабла выше, но и объем занимаемой памяти больше. 
API возвращает эти показатели с точностью до одной сотой, что умещается в гарантированные 6 знаков после запятой у Float.

Denis7God committed:
Добавлена проверка состояния сети на первом экране приложения; внесены коррективы в соответствии с комментариями к предыдущему pull request.

- На экране выбора региона (стартовый экран) добавлена проверка состояния подключения к сети. Для этого в проект добавлен файл "Reachability.swift", а в RegionsTableViewController при появлении и переходе реализована подписка/отписка на уведомления о состоянии в сети, обрабатываемая методом reachabilityChanged;
- Явно извлекаемые опционалы у переменных country и countryInfo в CountryInfoViewController заменены на обычные опционалы;
- try? инструкции заменены do(try)-catch блоками;
- Строки JSON полей моделях Country и CountryInfo вынесены во вложенные перечисления CodingKeys, а свойствам Country даны понятные имена.

OleksiiKarpenko commented on Atlas/Model/Country+CoreDataClass.swift:
```diff
@@ -9,19 +9,30 @@
+        guard let name = json["name"],
+              let nativeName = json["nativeName"],
+              let alpha2Code = json["alpha2Code"],
+              let alpha3Code = json["alpha3Code"]
```
Что за странные названия?

OleksiiKarpenko commented on Atlas/Controller/RegionsTableViewController.swift:
```diff
@@ -33,14 +34,44 @@ class RegionsTableViewController: UITableViewController {
+            try Reachability.shared.startNotifier()
+        }catch{
+            print("could not start reachability notifier")
+        }
```
Нет пробелов возле скобочек
